spec/services/dttp/retrieve_contact_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module Dttp
+  describe RetrieveContact do
+    describe "#call" do
+      let(:contact_entity_id) { SecureRandom.uuid }
+      let(:trainee) { create(:trainee, dttp_id: contact_entity_id) }
+      let(:filters) { "emailaddress1,firstname,lastname,contactid" }
+      let(:path) { "/contacts(#{trainee.dttp_id})?$select=#{filters}" }
+
+      before do
+        allow(AccessToken).to receive(:fetch).and_return("token")
+        allow(Client).to receive(:get).with(path).and_return(dttp_response)
+      end
+
+      context "filtered params" do
+        let(:parsed_response) do
+          {
+            "contactid" => "XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+            "firstname" => "John",
+            "lastname" => "Smith",
+          }
```

Comment:
missing `emailaddress1`
```suggestion
           "emailaddress1" => ""email@example.com,
          }
```